packages/scene-composer/src/components/panels/ComponentEditMenu.tsx

```diff
@@ -0,0 +1,168 @@
+});
+
+export const ComponentEditMenu: React.FC<ComponentEditMenuProps> = ({ nodeRef, currentComponent }) => {
+  const sceneComposerId = useContext(sceneComposerIdContext);
+  const updateComponentInternal = useStore(sceneComposerId)((state) => state.updateComponentInternal);
+  const removeComponent = useStore(sceneComposerId)((state) => state.removeComponent);
+  const { formatMessage } = useIntl();
+
+  const mapToMenuItem = useCallback(
+    (item: ToolbarItemOptionRaw): ComponentEditMenuItem => {
+      const typeId: ObjectTypes = item.uuid as ObjectTypes;
+      return {
+        ...item,
+        uuid: typeId,
+        subItems: item.subItems?.map(mapToMenuItem),
+        label: formatMessage(labelStrings[typeId]),
+        text: textStrings[typeId] ? formatMessage(textStrings[typeId]) : undefined, // if there's a text string, show text
+      };
+    },
+    [formatMessage],
+  );
+
+  const componentItems = useMemo(() => {
+    switch (currentComponent.type) {
+      case KnownComponentType.DataBinding:
+        return [
+          {
+            uuid: ObjectTypes.AddDataBinding,
+          },
+          {
+            uuid: ObjectTypes.RemoveAllDataBinding,
+          },
+        ];
+
+      case KnownComponentType.DataOverlay: {
+        const items = [
+          {
+            uuid: ObjectTypes.AddDataBinding,
+          },
+        ];
+        if ((currentComponent as IDataOverlayComponentInternal).subType === Component.DataOverlaySubType.OverlayPanel) {
+          items.push({
+            uuid: ObjectTypes.RemoveOverlay,
+          });
+        }
+        return items;
+      }
+      default:
+        return [];
+    }
+  }, [currentComponent]);
+
+  const menuItems = useMemo(
+    () =>
+      [
+        {
+          icon: { name: 'ellipsis' as IconProps.Name },
+          uuid: ObjectTypes.EditComponent,
+        },
+        ...componentItems,
+      ].map(mapToMenuItem),
+    [currentComponent],
+  );
+
+  const handleAddDataBinding = useCallback(() => {
+    switch (currentComponent.type) {
+      case KnownComponentType.DataBinding: {
+        const newComponentPartial = {
+          ...currentComponent,
+          valueDataBindings: [...((currentComponent as IDataBindingComponentInternal).valueDataBindings || []), {}],
+        };
+        updateComponentInternal(nodeRef, newComponentPartial);
+        return;
+      }
+      case KnownComponentType.DataOverlay: {
+        const newComponentPartial = {
+          ...currentComponent,
+          valueDataBindings: [
+            ...((currentComponent as IDataOverlayComponentInternal).valueDataBindings || []),
+            { bindingName: generateUUID() },
+          ],
+        };
+        updateComponentInternal(nodeRef, newComponentPartial);
+        return;
+      }
+      default:
+        throw `Add data binding not implemented for current component type ${currentComponent.type}`;
+    }
+  }, [nodeRef, currentComponent]);
```

Comment:
Non-blocker: This function could benefit from refactoring into small functions. It's a bit hard to read. Please consider improving this. Same comment applies to other functions this file.

Reply:
it seems to be fine currently that each function has a couple of switch cases with a couple of statements for different component types. we can see how to refactor when we need to add more cases

Reply:
there are some menu items shareable by multiple component types, and some items are component type specific. so I'd like to keep this for now and think about how to better refactor when we have more components migrating to use this menu